components/brave_ads/core/browser/service/network_client.cc

```diff
@@ -0,0 +1,145 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/browser/service/network_client.h"
+
+#include <cstddef>
+#include <utility>
+
+#include "base/containers/flat_map.h"
+#include "base/functional/bind.h"
+#include "base/strings/string_util.h"
+#include "brave/components/brave_ads/core/browser/service/network_client_util.h"
+#include "brave/components/brave_ads/core/mojom/brave_ads.mojom.h"
+#include "net/base/net_errors.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/mojom/network_context.mojom.h"
+#include "services/network/public/mojom/url_response_head.mojom.h"
+#include "url/gurl.h"
+
+namespace brave_ads {
+
+namespace {
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("brave_ads", R"(
+      semantics {
+        sender: "Brave Ads"
+        description:
+          "This service is used to communicate with Brave servers "
+          "to send and retrieve information for Ads."
+        trigger:
+          "Triggered by user viewing ads or at various intervals."
+        data:
+          "Ads catalog and Confirmations."
+        destination: WEBSITE
+      }
+      policy {
+        cookies_allowed: NO
+        setting:
+          "You can enable or disable this feature by visiting brave://rewards."
+        policy_exception_justification:
+          "Not implemented."
+      }
+    )");
```

Comment:
This is legacy, copied from AdsServiceImpl, but I’ll follow up with @fmarier and others to bring it up to date in a followup PR.